src/main/java/org/spongepowered/common/service/pagination/PaginationCalculator.java

```diff
@@ -85,88 +91,206 @@ public int getLinesPerPage(MessageReceiver source) {
-            int j = UNICODE_CHAR_WIDTHS[codePoint] >>> 4;
-            int k = UNICODE_CHAR_WIDTHS[codePoint] & 15;
+            width = 0;
+        }
+        //if bolded width gets 1 added.
+        if(isBold && width > 0) width = width + 1;
 
-            if (k > 7) {
-                k = 15;
-                j = 0;
-            }
-            width = ((k + 1) - j) / 2 + 1;
-            if (isBold) {
-                width += 0.5;
+        return width;
+    }
+
+    /**
+     * get width using legacy methods, unsure what the difference is.
+     * @param text
+     * @return the number of character pixels/columns the line takes up
+     */
+    private int getLegacyWidth(String text){
+        if (text == null) {
+            return 0;
+        } else {
+            int total = 0;
+            boolean bold = false;
+            for (int index = 0; index < text.length(); ++index) {
+                char charElement = text.charAt(index);
+                int width = this.getWidth(charElement, bold);
+                if (charElement == TextSerializers.LEGACY_FORMATTING_CODE.getCharacter() && index < text.length() - 1) {
+                    ++index;
+                    charElement = text.charAt(index);
+
+                    if (charElement == 'l' || charElement == 'L') {
+                        bold = true;
+                    } else {
+                        if (charElement == 'r' || charElement == 'R') {
+                            bold = false;
+                        }
+                    }
+                    width = 0;
+                }
+                total += width;
             }
+            return total;
         }
-        return width;
     }
 
-    private int getLength(MessageReceiver source, Text text) {
-        double columnCount = 0d;
-        for (Text child : text.withChildren()) {
-            final String txt;
-            if (child instanceof LiteralText) {
-                txt = ((LiteralText) child).getContent();
-            } else if (child instanceof TranslatableText) {
-                txt = child.toPlain();
+    /**
+     * compute the length of a given text
+     * @param text
+     * @return the number of character pixels/columns the line takes up
+     */
+    @VisibleForTesting
+    int getLength(Text text) {
+        //TODO: work out what the difference is between the legacy width, and getLength, and implement it in getLength.
+        if(true)
+            return getLegacyWidth(TextSerializers.LEGACY_FORMATTING_CODE.serialize(text));
+
+        IChatComponent component = SpongeTexts.toComponent(text);
+
+        Iterator<IChatComponent> it = component.iterator();
+        int total = 0;
+
+        while(it.hasNext()){
+            PrimitiveIterator.OfInt i_it;
+            IChatComponent child = it.next();
+            if(child instanceof ChatComponentText) {
+                i_it = ((ChatComponentText) child).getUnformattedTextForChat().codePoints().iterator();
```

Comment:
I originally thought that I was going to need to call 2 different methods, Thanks.